packages/astro/src/core/routing/match.ts

```diff
@@ -76,3 +77,18 @@ export function requestIs404Or500(request: Request, base = '') {
 
 	return isRoute404(pathname) || isRoute500(pathname);
 }
+
+/**
+ * Determines whether a given route is an external redirect.
+ *
+ * @param {RouteData} route - The route object to check.
+ * @return {boolean} Returns true if the route is an external redirect, otherwise false.
+ */
+export function isRouteExternalRedirect(route: RouteData): boolean {
+	if (route.type === 'redirect') {
+		if (route.redirect && redirectIsExternal(route.redirect)) {
+			return true;
+		}
+	}
+	return false;
```

Comment:
```suggestion
	return route.type === 'redirect' && route.redirect && redirectIsExternal(route.redirect);
```

Reply:
That's not the same thing because the binary expression might return `undefined` or `null`. The compiler catches this issue

Reply:
Changed in [`b2f9e55` (#13133)](https://github.com/withastro/astro/pull/13133/commits/b2f9e5559e5b1056bf3e622ee87701a6bcb5ba83)